packages/terminal/src/browser/terminal-info-toolbar-item.tsx

```diff
@@ -0,0 +1,113 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { ReactTabBarToolbarItem } from '@theia/core/lib/browser/shell/tab-bar-toolbar';
+import { ReactNode } from '@theia/core/shared/react';
+import React = require('@theia/core/shared/react');
+import { HoverService, Widget } from '@theia/core/lib/browser';
+import { TerminalWidget } from './base/terminal-widget';
+import { MarkdownRenderer } from '@theia/core/lib/browser/markdown-rendering/markdown-renderer';
+import { MarkdownStringImpl } from '@theia/core/lib/common/markdown-rendering/markdown-string';
+import { DisposableCollection } from '@theia/core';
+
+export class TerminalInfoToolbarItem implements ReactTabBarToolbarItem {
+    readonly id = 'terminal:info';
+
+    constructor(
+        protected readonly hoverService: HoverService,
+        protected readonly markdownRenderer: MarkdownRenderer
+    ) {}
+
+    isVisible(widget?: Widget): boolean {
+        return widget instanceof TerminalWidget;
+    }
+
+    render(widget?: Widget): ReactNode {
+        const toDispose = new DisposableCollection();
+        return (
+            <div
+                id={this.id}
+                className='codicon codicon-terminal-bash action-label'
+                onMouseEnter={e => this.onMouseEnter(e, toDispose, widget)}
+                onMouseLeave={e => this.onMouseLeave(toDispose)}
+            ></div>
+        );
+    }
+
+    protected async onMouseEnter(
+        event: React.MouseEvent<HTMLElement, MouseEvent>, toDispose: DisposableCollection, currentTerminal?: Widget
+    ): Promise<void> {
+        const currentTarget = event.currentTarget;
+        if (currentTerminal instanceof TerminalWidget) {
+            const extensions = await currentTerminal.contributingExtensions;
```

Comment:
Wouldn't it make more sense to let the TerminalWidget render (and cache) a Markdown string that we can just render with the markdow renderer instead of manipulating divs ourselves here?

Reply:
When adapting the enhanced preview this definitely makes sense. So I will change this.